pupil_src/player/gaze_producers.py

```diff
@@ -32,6 +32,23 @@
 gaze_mapping_plugins_by_name = {p.__name__: p for p in gaze_mapping_plugins}
 
 
+class Gaze_Producer_Base(Plugin):
+    uniqueness = 'by_base_class'
+    order = .02
+
+
+class Gaze_From_Recording(Gaze_Producer_Base):
+    def __init__(self, g_pool):
+        super().__init__(g_pool)
+        gaze_data_path = os.path.join(g_pool.rec_dir, "pupil_data")
```

Comment:
this is slow and takes a loot of RAM. Rather let the main.py script load pupil_data into g_pool.pupil_data and then access that from here.